src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedAutoPropAccessorSymbol.cs

```diff
@@ -0,0 +1,205 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Microsoft.Cci;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedAutoPropAccessorSymbol : SynthesizedInstanceMethodSymbol
+    {
+        public enum AccessorKind : byte
+        {
+            Get,
+            Init
+        }
+
+        private readonly SourcePropertySymbolBase _property;
+        private readonly AccessorKind _accessorKind;
+        private readonly TypeWithAnnotations.Boxed? _voidType;
```

Comment:
>TypeWithAnnotations.Boxed? [](start = 25, length = 26)

I do not think the use of boxed type is warranted here. There is no concurrency in initialization of this field, it is either set in the constructor or remains default. We have a reliable way to check for that - TypeWithAnnotations.HasType.